crates/red_knot/src/program/check.rs

```diff
@@ -39,7 +39,30 @@ impl Program {
     ) -> Result<Diagnostics, CheckError> {
         context.cancelled_ok()?;
 
-        // TODO schedule the dependencies.
+        let symbol_table = self.symbol_table(file);
+        let dependencies = symbol_table.dependencies();
+
+        if !dependencies.is_empty() {
+            let module_name = self.file_to_module(file).map(|module| module.name(self));
+
+            // TODO scheduling all dependencies here is wasteful if we don't infer any types on them
+            //  but I think that's unlikely, so it is okay?
+            //  Anyway, we need to figure out a way to retrieve the dependencies of a module
+            //  from the persistent cache. So maybe it should be a separate query after all.
```

Comment:
Will we need dependencies of a module in cases where we don't also need its symbol table anyway?

Reply:
Maybe not if we can short-circuit the entire `check_file` operation. I'll look into this next week